api/controllers/game/spectate.js

```diff
@@ -1,17 +1,37 @@
 module.exports = async function (req, res) {
   try {
     const { gameId } = req.body;
-    const game = await gameService.populateGame({ gameId });
+    const promiseGame = await gameService.populateGame({ gameId });
+    const promiseSpectator = await userService.findUser({ userId: req.session.usr });
```

Comment:
To @seriouslysean 's point below, we actually _don't_ want `await` on these lines because we _instead_ want to await the `Promise.all()` of both of them.

```suggestion
    const promiseGame = gameService.populateGame({ gameId });
    const promiseSpectator = userService.findUser({ userId: req.session.usr });
```

Reply:
That said, @seriouslysean's suggestion to inline them inside the Promise.all() call itself is reasonable imo. It gets rid of the variable declarations and shortens things up without being difficult to read because the actual calls themselves are pretty short. I say we go with that suggestion